pkg/envoy/xds_server.go

```diff
@@ -1574,17 +1574,22 @@ func getNetworkPolicy(ep endpoint.EndpointUpdater, vis *policy.VisibilityPolicy,
 		EndpointId:       ep.GetID(),
 		ConntrackMapName: ep.ConntrackNameLocked(),
 	}
-	// If no policy, deny all traffic. Otherwise, convert the policies for ingress and egress.
```

Comment:
What's the testing coverage like for this? How do we know this change doesn't introduce a network policy enforcement issue?

Reply:
This change allows policy to be sent to Envoy also when there is no policy (map) this was the case when policy was not enforced. Previously this did not matter because in such a case we never redirected any traffic to Envoy. Now, with Cilium Ingress (and L7 LB) in general, it is possible to redirect traffic to Envoy without policy being enforced. In this case we now need the "allow-all" policy to be sent to Envoy as otherwise it would drop all traffic now that we set the `EnforcePolicyOnL7Lb` flag on.

I did not add any testing for this yet, but had to add this change to be able to pass the current CI's Ingress tests. More specific Ingress policy test cases are planned to be added on the Cilium CLI test suite before the release.